### PR TITLE
fix: fit euler's totient to the contribution guidelines.

### PR DESCRIPTION
**Pull request by realstealthninja**
#### Description of Change
Made some minor adjustment to the algorithm itself 
by inverting the if statement.
Removed an unneccessary include.
Added tests.

#### Checklist

- [x] Added description of change
- [ ] Added file name matches [File name guidelines](https://github.com/TheAlgorithms/C-Plus-Plus/blob/master/CONTRIBUTING.md#New-File-Name-guidelines)
- [x] Added tests and example, test must pass
- [x] Added documentation so that the program is self-explanatory and educational - [Doxygen guidelines](https://www.doxygen.nl/manual/docblocks.html)
- [x] Relevant documentation/comments is changed or added
- [x] PR title follows semantic [commit guidelines](https://github.com/TheAlgorithms/C-Plus-Plus/blob/master/CONTRIBUTING.md#Commit-Guidelines)
- [x] Search previous suggestions before making a new one, as yours may be a duplicate.
- [x] I acknowledge that all my contributions will be made under the project's license.

Notes: Matched guidelines

### CODE REVIEW

File: math/eulers_totient_function.cpp [hunk 1]
```diff
@@ -1,58 +1,53 @@
...
- * Applying this 3 properties on the first equation.
- * \f[\phi(n) =
- * n\cdot\left(1-\frac{1}{p_1}\right)\cdot\left(1-\frac{1}{p_2}\right)\cdots\f]
- * where \f$p_1\f$,\f$p_2\f$... are prime factors.
- * Hence Implementation in \f$O\left(\sqrt{n}\right)\f$.
- * <br/>Some known values are:
- * * \f$\phi(100) = 40\f$
- * * \f$\phi(1) = 1\f$
- * * \f$\phi(17501) = 15120\f$
- * * \f$\phi(1420) = 560\f$
```
Comment (Panquesito7) [inline math/eulers_totient_function.cpp:25]: I think this description was totally fine. Please add it back.

File: math/eulers_totient_function.cpp [hunk 2]
```diff
@@ -1,58 +1,53 @@
...
  */
-#include <cstdlib>
-#include <iostream>
 
-/** Function to caculate Euler's totient phi
+#include <iostream> // for io operations
+#include <cassert> // for assert
+
+namespace math {
+
```
Comment (Panquesito7) [inline math/eulers_totient_function.cpp:15]:
```suggestion
/**
 * @brief Mathematical algorithms
 * @namespace
 */
namespace math {
```

File: math/eulers_totient_function.cpp [hunk 3]
```diff
@@ -1,58 +1,53 @@
...
+        result -= result / i;
     }
-    if (n > 1)
-        result -= result / n;
+    if (n > 1) result -= result / n;
+
     return result;
 }
+} // namespace math
```
Comment (Panquesito7) [inline math/eulers_totient_function.cpp:32]:
```suggestion
}  // namespace math
```

File: math/eulers_totient_function.cpp [hunk 4]
```diff
@@ -1,58 +1,53 @@
...
+
+/**
+ * @brief Self-test implementations
+ * @returns void
+ */
+static void test() {
+    assert(math::phiFunction(1) == 1);
+    assert(math::phiFunction(2) == 1);
+    assert(math::phiFunction(10) == 4);
```
Comment (Panquesito7) [inline math/eulers_totient_function.cpp:41]: It'd be great if one or more assert tests could be added. 🙂

**Review by Panquesito7 — CHANGES_REQUESTED** [review]

File: math/eulers_totient_function.cpp (hunk 1 above)
Comment (realstealthninja) [reply math/eulers_totient_function.cpp:25]: i dont know how thats readable but ill do that

File: math/eulers_totient_function.cpp (hunk 4 above)
Comment (realstealthninja) [reply math/eulers_totient_function.cpp:44]: your wish is my command :smile:

File: math/eulers_totient_function.cpp (hunk 1 above)
Comment (realstealthninja) [reply math/eulers_totient_function.cpp]: i think the same

File: math/eulers_totient_function.cpp [hunk 5]
```diff
@@ -23,36 +21,56 @@
...
  * * \f$\phi(17501) = 15120\f$
  * * \f$\phi(1420) = 560\f$
+ * @author [Mann Mehta](https://github.com/mann2108)
  */
-#include <cstdlib>
-#include <iostream>
 
-/** Function to caculate Euler's totient phi
+#include <iostream> // for io operations
+#include <cassert> // for assert
```
Comment (Panquesito7) [inline math/eulers_totient_function.cpp:28]:
```suggestion
#include <iostream> /// for IO operations
#include <cassert> ///for assert
```

File: math/eulers_totient_function.cpp [hunk 6]
```diff
@@ -23,36 +21,56 @@
...
+
+/**
+ * @brief Mathematical algorithms
+ * @namespace
+ */
+namespace math {
+/**
+ * @brief Function to calculate Euler's Totient
+ * @param n is the number to find the Euler's Totient of.
```
Comment (Panquesito7) [inline math/eulers_totient_function.cpp:37]:
```suggestion
 * @param n the number to find the Euler's Totient of
```

File: math/eulers_totient_function.cpp [hunk 7]
```diff
@@ -23,36 +21,56 @@
...
  * * \f$\phi(17501) = 15120\f$
  * * \f$\phi(1420) = 560\f$
+ * @author [Mann Mehta](https://github.com/mann2108)
  */
-#include <cstdlib>
-#include <iostream>
 
-/** Function to caculate Euler's totient phi
+#include <iostream> /// for IO operations
+#include <cassert> ///for assert
```
Comment (Panquesito7) [inline math/eulers_totient_function.cpp:28]:
```suggestion
#include <cassert> /// for assert
```

File: math/eulers_totient_function.cpp [hunk 8]
```diff
@@ -23,36 +21,56 @@
...
+ * @returns void
+ */
+static void test() {
+    assert(math::phiFunction(1) == 1);
+    assert(math::phiFunction(2) == 1);
+    assert(math::phiFunction(10) == 4);
+    assert(math::phiFunction(123456) == 41088);
+    assert(math::phiFunction(808017424794) == 263582333856);
+    assert(math::phiFunction(3141592) == 1570792);
+    assert(math::phiFunction(27182818) == 12545904);
```
Comment (Panquesito7) [inline math/eulers_totient_function.cpp:64]:
```suggestion
    assert(math::phiFunction(27182818) == 12545904);

    std::cout << "All tests have successfully passed!\n";
```

**Review by Panquesito7 — APPROVED** [review]
Awesome. Thanks! 🚀

**Review by tjgurwara99 — APPROVED** [review]

**Pull request merged**